bsmetadata/preprocessing_utils.py

```diff
@@ -171,7 +171,9 @@ def _extract_website_desc_from_url(self, url: str) -> Optional:
 class EntityPreprocessor(MetadataPreprocessor):
     """Metadata preprocessor for adding entity information."""
 
-    def __init__(self, base_url):
+    def __init__(self, base_url, path_wiki_db):
+        self.wiki_db_path = path_wiki_db
+        self.entity_utils = WikipediaDescUtils(path_wiki_db)
```

Comment:
There's a small concern that I have here. Calling the init method of WikipediaDescUtils from two different processors would initialize the `dump_db `and `redirects_map` twice. While not being ideal design wise, this would also add up to the time since `redirects_map`, which takes some substantial time to be populated. would be created from scratch again. Instead, we can have it at a higher level (maybe at Metadata Processor level) or find another way, maybe cache or something which initializes this only once. cc: @SaulLu @timoschick

Reply:
Yes, absolutely. We can have a separate PR to handle this in the best manner :)